### DISCUSSION TITLE
shelly 2.5 cycle switch

### DISCUSSION THREAD

**Discussion by kx3kx3kx3** [mongoose-os-apps/shelly-homekit#1455] [Ideas]
it's possible to reproduce this with homekit 
![IMG_0138](https://user-images.githubusercontent.com/58489045/96021260-7995d580-0e4f-11eb-8765-59287c2b4ff5.jpg)

**Comment by rojer** [comment]
i guess. does it mean tapping either tile in homekit does this?

**Comment by andyblac** [comment]
> i guess. does it mean tapping either tile in homekit does this?

i think only for physical switch. for single switch to use on 2 channels

**Comment by rojer** [comment]
either way with the way code is structured now, it would need some ugly hacks, as currently switches and their inputs are completely independent.
not impossible, but not trivial. let's see if there's more demand for this.

**Comment by kx3kx3kx3** [comment]
> > i guess. does it mean tapping either tile in homekit does this?
> 
> i think only for physical switch. for single switch to use on 2 channels

yes only for physical push button

**Comment by rojer** [comment]
with just the physical button, it's a bit simpler but still, kinda meh. will leave this one to gather more support.

**Comment by riccardog821** [comment]
Up 
I have installed HomeKit firmware on a Shelly 2.5 that was set as cycle switch controlling 3 lights with 2 control buttons. Now there is a way about how configure the Shelly so that pressing first time the button “channel 1” will be turned on and on a second push the “channel2” will be on?

**Comment by kx3kx3kx3** [comment]
I found how to do it but it is an old firmware that is no longer supported

**Comment by riccardog821** [comment]
Oh what a sad news :( so no any other option to make it work as per the original firmware but also on HomeKit?

**Comment by lucaspinelli85** [comment]
I think it can be useful to many

**Comment by lucaspinelli85** [comment]
In this case the relay would no longer be directly connected to the button and would only work via the wifi network. I don't want to avoid disconnection problems, if there is no wifi network nothing would work. It would be much better if the function is integrated into the device, so it works even if it is not connected in wifi. Can you put it in the 3.0 firmware?

**Comment by lucaspinelli85** [comment]
when will the cycle switch function be implemented?![image](https://user-images.githubusercontent.com/74553551/147698842-5d0a1711-d90d-4d19-b697-e638d71e83ef.jpeg)

**Comment by lucaspinelli85** [comment]
Please @rojer can you implement this feature?
In two rooms I only have space for one button for two lamps, I can't add another button ... Currently I have the original firmware and I would like to transform it HomeKit like all the other shellys I have!
Thank you!

**Comment by andreagv369** [comment]
Yes, please! This is an important function in Shelly 2.5, but most  of all in Plus 2PM

**Comment by robfero** [comment]
Please implement this. Your firmware is solving a lot of issues I had with homebridge. This is the only thing that I'm missing.

**Comment by antodan91** [comment]
I follow this discussion and I hope that this feature will be implemented. @rojer do you think that it is possibile? Many thanks for your firmware.

**Comment by Tomvol-1** [comment]
Ciao , è possibile implementare questa opzione?

Grazie
